vendor/k8s.io/kubernetes/pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1499,6 +1522,28 @@ func (d *awsDisk) describeVolume() (*ec2.Volume, error) {
 	return volumes[0], nil
 }
 
+// applyUnSchedulableTaint applies a unschedulable taint to a node after verifying
+// if node has become unusable because of volumes getting stuck in attaching state.
+func (c *Cloud) applyUnSchedulableTaint(nodeName types.NodeName, reason string) {
+	node, fetchErr := c.kubeClient.CoreV1().Nodes().Get(string(nodeName), metav1.GetOptions{})
+	if fetchErr != nil {
+		glog.Errorf("Error fetching node %s with %v", nodeName, fetchErr)
+		return
+	}
+
+	taint := &v1.Taint{
+		Key:    nodeWithImpairedVolumes,
+		Value:  "true",
+		Effect: v1.TaintEffectNoSchedule,
+	}
+	err := controller.AddOrUpdateTaintOnNode(c.kubeClient, string(nodeName), taint)
```

Comment:
Is "cloud-provider" allowed to get and update Node objects?

Reply:
Good catch. I have opened a follow up PR for fixing cloud-provider's access to node objects. https://github.com/kubernetes/kubernetes/pull/56709 

Originally when I tested this, I was running a cluster without RBAC. :(